## Analysis of Men's and Women's 2023 NCAA DI Swimming Championships

I will analyze the results of the 2023 NCAA championship meet and answer a few questions:
1. Which team(s) did best (and how should that be defined)?
2. Are there any differences in performance based on a swimmer's year?
3. What are the average improvements from qualifying times to preliminaries to finals? Do any factors impact this?

### Loading and Inspecting Data

Most of this step was handled in `buildData_2023NCAAs.ipynb` where I converted PDF results to a CSV file. I'll load the CSVs and do some inspection and any small cleaning required.

In [1]:
import pandas as pd
import numpy as np

In [2]:
# Load the men's and women's results and combine dataframes
dfResultsM = pd.read_csv('NCAA_M2023.csv')
dfResultsW = pd.read_csv('NCAA_W2023.csv')
dfResultsM['Division'] = 'Men\'s DI'
dfResultsW['Division'] = 'Women\'s DI'
dfResults = pd.concat([dfResultsM, dfResultsW], ignore_index=True)
dfResults.head()

Unnamed: 0                                Event           Category  \
0           0   Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
1           1   Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
2           2   Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
3           3   Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
4           4   Event 1  Men 200 Yard Medley Relay  Timed Final Relay   

               Name1 Year1              Name2 Year2            Name3 Year3  \
0  Stokowski, Kacper    SR      Hunter, Mason    5Y  Korstanje, Nyls    SR   
1        Dolan, Jack    SR     Marchand, Leon    SO    McCusker, Max    5Y   
2       Chaney, Adam    JR  Savickas, Aleksas    FR     Friese, Eric    SR   
3    Seeliger, Bjorn    JR         Bell, Liam    SR       Rose, Dare    JR   
4     Burns, Brendan    SR       Mathias, Van    5Y   Frankel, Tomer    JR   

            Name4 Year4      School QualifyingTime     Time Place  Points  \
0  Curtiss, David    SO    NC State        1:22.25  1:20.67     1    40.0   
1    Kulow, Jonny    FR  Arizona St        1:21.69  1:21.07     2    34.0   
2    Liendo, Josh    FR     Florida        1:21.73  1:21.14     3    32.0   
3     Alexy, Jack    SO  California        1:22.84  1:21.24     4    30.0   
4    Wight, Gavin    JR     Indiana        1:23.52  1:21.52     5    28.0   

   Division  
0  Men's DI  
1  Men's DI  
2  Men's DI  
3  Men's DI  
4  Men's DI

In [3]:
dfResults.describe(include='all')

Unnamed: 0                               Event       Category  \
count   1949.000000                                1949           1949   
unique          NaN                                  39              6   
top             NaN   Event 3  Women 500 Yard Freestyle  Preliminaries   
freq            NaN                                  84           1255   
mean     488.628014                                 NaN            NaN   
std      284.614824                                 NaN            NaN   
min        0.000000                                 NaN            NaN   
25%      243.000000                                 NaN            NaN   
50%      487.000000                                 NaN            NaN   
75%      730.000000                                 NaN            NaN   
max     1034.000000                                 NaN            NaN   

                     Name1 Year1        Name2 Year2         Name3 Year3  \
count                 1949  1949          229   229           229   229   
unique                 547     5          164     5           164     5   
top     Berkoff, Katharine    SR  Walsh, Alex    SR  Jones, Emily    SR   
freq                    10   495            4    61             4    68   
mean                   NaN   NaN          NaN   NaN           NaN   NaN   
std                    NaN   NaN          NaN   NaN           NaN   NaN   
min                    NaN   NaN          NaN   NaN           NaN   NaN   
25%                    NaN   NaN          NaN   NaN           NaN   NaN   
50%                    NaN   NaN          NaN   NaN           NaN   NaN   
75%                    NaN   NaN          NaN   NaN           NaN   NaN   
max                    NaN   NaN          NaN   NaN           NaN   NaN   

              Name4 Year4   School QualifyingTime  Time Place       Points  \
count           229   229     1949           1949  1949  1949  1949.000000   
unique          168     5       70           1662  1702    69          NaN   
top     Arens, Abby    SO  Florida          51.90   DFS   ---          NaN   
freq              4    54      126              6    43    66          NaN   
mean            NaN   NaN      NaN            NaN   NaN   NaN     3.658286   
std             NaN   NaN      NaN            NaN   NaN   NaN     7.421645   
min             NaN   NaN      NaN            NaN   NaN   NaN     0.000000   
25%             NaN   NaN      NaN            NaN   NaN   NaN     0.000000   
50%             NaN   NaN      NaN            NaN   NaN   NaN     0.000000   
75%             NaN   NaN      NaN            NaN   NaN   NaN     4.000000   
max             NaN   NaN      NaN            NaN   NaN   NaN    40.000000   

          Division  
count         1949  
unique           2  
top     Women's DI  
freq          1035  
mean           NaN  
std            NaN  
min            NaN  
25%            NaN  
50%            NaN  
75%            NaN  
max            NaN

Everything looks ok after reading the two dataframes in, except as I wrote/read I ended up with 2 index columns, so I can drop the 'Unnamed: 0' column.

In [4]:
dfResults = dfResults.drop(['Unnamed: 0'], axis=1)

In [5]:
dfResults.head(25)

Event           Category  \
0       Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
1       Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
2       Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
3       Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
4       Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
5       Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
6       Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
7       Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
8       Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
9       Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
10      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
11      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
12      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
13      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
14      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
15      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
16      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
17      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
18      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
19      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
20      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
21      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
22      Event 1  Men 200 Yard Medley Relay  Timed Final Relay   
23   Event 2  Men 800 Yard Freestyle Relay  Timed Final Relay   
24   Event 2  Men 800 Yard Freestyle Relay  Timed Final Relay   

                      Name1 Year1                      Name2 Year2  \
0         Stokowski, Kacper    SR              Hunter, Mason    5Y   
1               Dolan, Jack    SR             Marchand, Leon    SO   
2              Chaney, Adam    JR          Savickas, Aleksas    FR   
3           Seeliger, Bjorn    JR                 Bell, Liam    SR   
4            Burns, Brendan    SR               Mathias, Van    5Y   
5           Kammann, Bjoern    SO            Houlie, Michael    5Y   
6              Lowe, Dalton    JR           Petrashov, Denis    SR   
7          Brownstead, Matt    JR              Nichols, Noah    JR   
8            Stoffle, Aidan    SR               Mikuta, Reid    JR   
9          MacAlister, Leon    SR              Polonsky, Ron    SO   
10              Menke, Matt    SR                Maas, Derek    SR   
11           Foster, Carson    JR                 Chan, Will    5Y   
12          Dunham, Bradley    SR            Pitshugin, Kris    FR   
13          Gogulski, Ethan    SR  Puente Bustamante, Andres    SR   
14               Jekel, Wes    SR                Myhre, Will    SR   
15              Quach, Alex    SO              Helmuth, Karl    SO   
16            Janton, Tommy    FR              Faikish, Sean    JR   
17             Davis, Wyatt    JR       Fitzpatrick, Brendan    SO   
18             Stowe, Casey    JR                McHugh, Max    5Y   
19  Radziszewski, Krzysztof    SO       Van Der Laan, Cooper    SR   
20              Grant, Will    JR             Simpson, Jared    SR   
21             Storms, Eric    5Y                Patton, Ben    SR   
22             Webb, Forest    SR                  Pouch, AJ    SR   
23             Hobson, Luke    SO             Carrozza, Coby    JR   
24             House, Grant    5Y            Sammon, Patrick    SO   

                   Name3 Year3              Name4 Year4         School  \
0        Korstanje, Nyls    SR     Curtiss, David    SO       NC State   
1          McCusker, Max    5Y       Kulow, Jonny    FR     Arizona St   
2           Friese, Eric    SR       Liendo, Josh    FR        Florida   
3             Rose, Dare    JR        Alexy, Jack    SO     California   
4         Frankel, Tomer    JR       Wight, Gavin    JR        Indiana   
5         Crooks, Jordan    SO  Santos, Guilherme    FR      Tennessee   
6   Ela

In [6]:
dfResults[dfResults.Category == 'Championship Final'].head()

Event            Category            Name1  \
44   Event 3  Men 500 Yard Freestyle  Championship Final     Hobson, Luke   
45   Event 3  Men 500 Yard Freestyle  Championship Final  Johnston, David   
46   Event 3  Men 500 Yard Freestyle  Championship Final    Magahey, Jake   
47   Event 3  Men 500 Yard Freestyle  Championship Final    Newmark, Jake   
48   Event 3  Men 500 Yard Freestyle  Championship Final   Mitchell, Jake   

   Year1 Name2 Year2 Name3 Year3 Name4 Year4     School QualifyingTime  \
44    SO   NaN   NaN   NaN   NaN   NaN   NaN      Texas        4:10.51   
45    JR   NaN   NaN   NaN   NaN   NaN   NaN      Texas        4:10.02   
46    JR   NaN   NaN   NaN   NaN   NaN   NaN    Georgia        4:10.83   
47    JR   NaN   NaN   NaN   NaN   NaN   NaN  Wisconsin        4:10.80   
48    JR   NaN   NaN   NaN   NaN   NaN   NaN    Florida        4:11.65   

       Time Place  Points  Division  
44  4:07.37     1    20.0  Men's DI  
45  4:08.79     2    17.0  Men's DI  
46  4:09.24     3    16.0  Men's DI  
47  4:10.12     4    15.0  Men's DI  
48  4:10.54     5    14.0  Men's DI

Very brief checks by eye of results for both a relay and an individual event don't show obvious major errors. I'll quickly modify the name columns to get First + Last names.

In [7]:
name = pd.DataFrame()
for col in ['Name1','Name2','Name3','Name4']:
    name[['Last', 'First']] = dfResults[col].str.split(',',expand=True)
    dfResults[col] = name['First'] + ' ' + name['Last']
dfResults.head()

Event           Category              Name1  \
0   Event 1  Men 200 Yard Medley Relay  Timed Final Relay   Kacper Stokowski   
1   Event 1  Men 200 Yard Medley Relay  Timed Final Relay         Jack Dolan   
2   Event 1  Men 200 Yard Medley Relay  Timed Final Relay        Adam Chaney   
3   Event 1  Men 200 Yard Medley Relay  Timed Final Relay     Bjorn Seeliger   
4   Event 1  Men 200 Yard Medley Relay  Timed Final Relay      Brendan Burns   

  Year1              Name2 Year2            Name3 Year3           Name4 Year4  \
0    SR       Mason Hunter    5Y   Nyls Korstanje    SR   David Curtiss    SO   
1    SR      Leon Marchand    SO     Max McCusker    5Y     Jonny Kulow    FR   
2    JR   Aleksas Savickas    FR      Eric Friese    SR     Josh Liendo    FR   
3    JR          Liam Bell    SR        Dare Rose    JR      Jack Alexy    SO   
4    SR        Van Mathias    5Y    Tomer Frankel    JR     Gavin Wight    JR   

       School QualifyingTime     Time Place  Points  Division  
0    NC State        1:22.25  1:20.67     1    40.0  Men's DI  
1  Arizona St        1:21.69  1:21.07     2    34.0  Men's DI  
2     Florida        1:21.73  1:21.14     3    32.0  Men's DI  
3  California        1:22.84  1:21.24     4    30.0  Men's DI  
4     Indiana        1:23.52  1:21.52     5    28.0  Men's DI

In [8]:
dfResults.dtypes

Event              object
Category           object
Name1              object
Year1              object
Name2              object
Year2              object
Name3              object
Year3              object
Name4              object
Year4              object
School             object
QualifyingTime     object
Time               object
Place              object
Points            float64
Division           object
dtype: object

Looking at the data types, I see everything is an object other than the points (there are some half point results in the case of a tie, so float is required for those). The Place could be an int, although currently non-placing swimmers are given a place of '---'. This could be made to be NaN so that the type could be changed to int. The times -- other than 'DQ' (disqualified) and 'DFS' (declared false start) -- could be stored as datetimes to make operations with them easier. Everythig else should stay an object.

In [9]:
dfResults.Event.unique()

array([' Event 1  Men 200 Yard Medley Relay',
       ' Event 2  Men 800 Yard Freestyle Relay',
       ' Event 3  Men 500 Yard Freestyle', ' Event 4  Men 200 Yard IM',
       ' Event 5  Men 50 Yard Freestyle',
       ' Event 7  Men 200 Yard Freestyle Relay',
       ' Event 8  Men 400 Yard IM', ' Event 9  Men 100 Yard Butterfly',
       ' Event 10  Men 200 Yard Freestyle',
       ' Event 11  Men 100 Yard Breaststroke',
       ' Event 12  Men 100 Yard Backstroke',
       ' Event 14  Men 400 Yard Medley Relay',
       ' Event 15  Men 1650 Yard Freestyle',
       ' Event 16  Men 200 Yard Backstroke',
       ' Event 17  Men 100 Yard Freestyle',
       ' Event 18  Men 200 Yard Breaststroke',
       ' Event 19  Men 200 Yard Butterfly',
       ' Event 21  Men 400 Yard Freestyle Relay',
       ' Event 110  Men 200 Yard Freestyle Swim-off',
       ' Event 112  Men 100 Yard Backstroke Swim-off',
       ' Event 119  Men 200 Yard Butterfly Swim-off',
       ' Event 1  Women 200 Yard Medley Relay',
 

In [10]:
# Get rid of extra leading/trailing spaces in string variables
# There's an extra space between event number and event name that is annoying but I'll leave as is for now
dfObj = dfResults.select_dtypes(['object'])
dfResults[dfObj.columns] = dfObj.apply(lambda x: x.str.strip())
dfResults.Event.unique()

array(['Event 1  Men 200 Yard Medley Relay',
       'Event 2  Men 800 Yard Freestyle Relay',
       'Event 3  Men 500 Yard Freestyle', 'Event 4  Men 200 Yard IM',
       'Event 5  Men 50 Yard Freestyle',
       'Event 7  Men 200 Yard Freestyle Relay',
       'Event 8  Men 400 Yard IM', 'Event 9  Men 100 Yard Butterfly',
       'Event 10  Men 200 Yard Freestyle',
       'Event 11  Men 100 Yard Breaststroke',
       'Event 12  Men 100 Yard Backstroke',
       'Event 14  Men 400 Yard Medley Relay',
       'Event 15  Men 1650 Yard Freestyle',
       'Event 16  Men 200 Yard Backstroke',
       'Event 17  Men 100 Yard Freestyle',
       'Event 18  Men 200 Yard Breaststroke',
       'Event 19  Men 200 Yard Butterfly',
       'Event 21  Men 400 Yard Freestyle Relay',
       'Event 110  Men 200 Yard Freestyle Swim-off',
       'Event 112  Men 100 Yard Backstroke Swim-off',
       'Event 119  Men 200 Yard Butterfly Swim-off',
       'Event 1  Women 200 Yard Medley Relay',
       'Event 2  Women 

In [11]:
dfResults[dfResults.Event == 'Event 1  Women 200 Yard Medley Relay']

Event           Category  \
914  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
915  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
916  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
917  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
918  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
919  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
920  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
921  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
922  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
923  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
924  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
925  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
926  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
927  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
928  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
929  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
930  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
931  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
932  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
933  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
934  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
935  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
936  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   
937  Event 1  Women 200 Yard Medley Relay  Timed Final Relay   

                           Name1 Year1                Name2 Year2  \
914               Gretchen Walsh    SO           Alex Walsh    JR   
915            Katharine Berkoff    SR  Heather MacCausland    SR   
916                  Olivia Bray    JR          Anna Elendt    JR   
917             Nyah Funderburke    SO                  NaN    SR   
918                     Abby Hay    SR       Cecilia Viberg    FR   
919             Isabelle Stadden    JR           Jade Neser    JR   
920                  Rhyan White    5Y        Avery Wiseman    SO   
921               Greer Pattison    SO         Skyler Smith    SO   
922                Claire Curzan    FR           Allie Raab    5Y   
923              Caroline Famous    JR       Kaitlyn Dobler    JR   
924             Josephine Fuller    SO        Mona McSharry    JR   
925               Anna Peplowski    SO     Noelle Peplowski    5Y   
926                Eboni McCarty    SO         Zoie Hartman    SR   
927              Noelle Kaufmann    JR          Letitia Sim    SO   
928  Andrea Sansores De La Fuent    5Y    Alessia Ferraguti    5Y   
929                Emma Shuppert    SR       Kaelyn Gridley    FR   
930              Tania Quaglieri    SR        Julia Mansson    FR   
931              Justine Murdock    SO      Hannah Brunzell    SR   
932                   Meghan Lee    JR      Stasya Makarova    SO   
933                Meredith Rees    SR         Malin Grosse    FR   
934              Sophia Kosturos    SR          Eva Carlson    SO   
935               Caitlin Brooks    SR        Denise Phelan    FR   
936              Jordan Buechler    JR      Andrea Perttula    SR   
937                 Aris Runnels    SO        Nina Kucheran    5Y   

                    Name3 Year3             Name4 Year4               School  \
914            Lexi Cuomo    SR     Kate Douglass    SR             Virginia   
915           Kylee Alons    5Y        Abby Arens    JR             NC State   
916         Emma Sticklen    JR      Grace Cooper    JR                Texas   
917                   NaN    JR       Teresa Ivan    SO              Ohio St   
918  Christiana Regenauer    SR      Gabi Albiero    JR           Louisville   
919             Mia Kragh    SO     Emma Davidson    SR           California   
920           Emily Jones    FR    Kalia Antoniou    5Y              Alabama   
921         Ellie VanNote    SR     Grace Countie    SR              

There are a few relay entries where the swimmer name columns dropped last names so ended up with NaN values after some manipulations. Since there are so few and I cannot see any reason why this would happen for just these handful so I assume it was a PDF reader error, I will just consult the results and fix these by hand.

In [12]:
dfResults.loc[917, ['Name2','Name3']] = ['Hannah Bach', 'Katherine Zenick']

In [13]:
dfResults[dfResults.Event == 'Event 14  Men 400 Yard Medley Relay']

Event           Category  \
577  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
578  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
579  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
580  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
581  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
582  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
583  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
584  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
585  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
586  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
587  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
588  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
589  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
590  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
591  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
592  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
593  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
594  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
595  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
596  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
597  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   
598  Event 14  Men 400 Yard Medley Relay  Timed Final Relay   

                      Name1 Year1                     Name2 Year2  \
577             Adam Chaney    JR             Dillon Hillis    5Y   
578           Brendan Burns    SR              Josh Matheny    SO   
579              Jack Dolan    SR             Leon Marchand    SO   
580        Kacper Stokowski    SR              Mason Hunter    5Y   
581            Destin Lasco    JR             Reece Whitley    5Y   
582             Nick Simons    FR        Lyubomir Epitropov    5Y   
583             Forest Webb    SR         Carles Coll Marti    JR   
584           Carson Foster    JR            Caspar Corbeau    SR   
585           Jack Dahlgren    5Y                Ben Patton    SR   
586         Matt Brownstead    JR              Noah Nichols    JR   
587         Leon MacAlister    SR              Ron Polonsky    SO   
588            Nate Stoffle    SO               Reid Mikuta    JR   
589          Ethan Gogulski    SR  Andres Puente Bustamante    SR   
590             Wyatt Davis    JR       Brendan Fitzpatrick    SO   
591               Wes Jekel    SR                Will Myhre    SR   
592  Krzysztof Radziszewski    SO       Cooper Van Der Laan    SR   
593              Will Grant    JR             Jared Simpson    SR   
594              Matt Menke    SR                Derek Maas    SR   
595              Alex Quach    SO              Karl Helmuth    SO   
596          Bradley Dunham    SR            Kris Pitshugin    FR   
597           Charlie Crush    FR           Denis Petrashov    SR   
598            Tommy Janton    FR                       NaN    JR   

                 Name3 Year3                   Name4 Year4         School  \
577        Josh Liendo    FR         Macguire McDuff    SO        Florida   
578      Tomer Frankel    JR         Rafael Miroslaw    SO        Indiana   
579       Max McCusker    5Y             Jonny Kulow    FR     Arizona St   
580        Aiden Hayes    SO             Luke Miller    JR       NC State   
581       Gabriel Jett    SO          Bjorn Seeliger    JR     California   
582      Jordan Crooks    SO        Guilherme Santos    FR      Tennessee   
583    Youssef Ramadan    JR  Luis Dominguez Calonge    SO  Virginia Tech   
584     Sterling Crane    SR             Luke Hobson    SO          Texas   
585     Clement Secchi    5Y         Grant Bochenski    SO       Missouri   
586        Tim Connery    SO             Jack Aikins    SO       Virginia   
587     Andrei Minakov    JR             Luke Maurer    JR       Stanford   
588      Aidan Stoffle    SR           Kalle Makinen    FR         Auburn

In [14]:
dfResults.loc[598, ['Name2','Name3','Name4']] = ['Sean Faikish', 'Cason Wilburn', 'Thacher Scannell']

In [15]:
dfResults[(dfResults.Name3.isna()) & (~dfResults.Year3.isna())]

Event           Category           Name1  \
935  Event 1  Women 200 Yard Medley Relay  Timed Final Relay  Caitlin Brooks   

    Year1          Name2 Year2 Name3 Year3 Name4 Year4    School  \
935    SR  Denise Phelan    FR   NaN    FR   NaN    SR  Kentucky   

    QualifyingTime     Time Place  Points    Division  
935        1:36.43  1:37.55    22     0.0  Women's DI

In [16]:
dfResults.loc[935, ['Name3','Name4']] = ['Lydia Hanlon', 'Kaitlynn Wheeler']

In [17]:
dfResults[(dfResults.Name4.isna()) & (~dfResults.Year4.isna())]

Empty DataFrame
Columns: [Event, Category, Name1, Year1, Name2, Year2, Name3, Year3, Name4, Year4, School, QualifyingTime, Time, Place, Points, Division]
Index: []

In [18]:
dfResults.describe(include='all')

Event       Category              Name1  \
count                                1949           1949               1949   
unique                                 39              6                547   
top     Event 3  Women 500 Yard Freestyle  Preliminaries  Katharine Berkoff   
freq                                   84           1255                 10   
mean                                  NaN            NaN                NaN   
std                                   NaN            NaN                NaN   
min                                   NaN            NaN                NaN   
25%                                   NaN            NaN                NaN   
50%                                   NaN            NaN                NaN   
75%                                   NaN            NaN                NaN   
max                                   NaN            NaN                NaN   

       Year1       Name2 Year2             Name3 Year3       Name4 Year4  \
count   1949         229   229               229   229         229   229   
unique     5         163     5               162     5         166     5   
top       SR  Alex Walsh    SR  Callie Dickinson    SR  Abby Arens    SO   
freq     495           4    61                 4    68           4    54   
mean     NaN         NaN   NaN               NaN   NaN         NaN   NaN   
std      NaN         NaN   NaN               NaN   NaN         NaN   NaN   
min      NaN         NaN   NaN               NaN   NaN         NaN   NaN   
25%      NaN         NaN   NaN               NaN   NaN         NaN   NaN   
50%      NaN         NaN   NaN               NaN   NaN         NaN   NaN   
75%      NaN         NaN   NaN               NaN   NaN         NaN   NaN   
max      NaN         NaN   NaN               NaN   NaN         NaN   NaN   

         School QualifyingTime  Time Place       Points    Division  
count      1949           1949  1949  1949  1949.000000        1949  
unique       70           1662  1702    69          NaN           2  
top     Florida          51.90   DFS   ---          NaN  Women's DI  
freq        126              6    43    66          NaN        1035  
mean        NaN            NaN   NaN   NaN     3.658286         NaN  
std         NaN            NaN   NaN   NaN     7.421645         NaN  
min         NaN            NaN   NaN   NaN     0.000000         NaN  
25%         NaN            NaN   NaN   NaN     0.000000         NaN  
50%         NaN            NaN   NaN   NaN     0.000000         NaN  
75%         NaN            NaN   NaN   NaN     4.000000         NaN  
max         NaN            NaN   NaN   NaN    40.000000         NaN

In [28]:
# Want to use datetime for the times, but there are some with minute values and some without
# This will convert all times with minutes, then any that don't work will be converted as seconds
# There are entries for DQs and DFSs that will be coerced into NaT.
# The default is for the date to be Jan. 1, 1900, which I will leave as is. I'll only worry about
# differences in the time pieces anyway and I can use a label to differentiate meets rather than their
# dates anyway if I need to compare multiple meets (plus the meets are multi-day so it just becomes complicated)
dfResults.Time = pd.to_datetime(dfResults.Time, format='%M:%S.%f', errors='coerce').fillna(
                pd.to_datetime(dfResults.Time, format='%S.%f', errors='coerce'))
dfResults.QualifyingTime = pd.to_datetime(dfResults.QualifyingTime, format='%M:%S.%f', errors='coerce').fillna(
                pd.to_datetime(dfResults.QualifyingTime, format='%S.%f', errors='coerce'))
dfResults.describe(datetime_is_numeric=True)

QualifyingTime                           Time  \
count                           1949                           1883   
mean   1900-01-01 00:02:25.066541568  1900-01-01 00:02:27.072899328   
min       1900-01-01 00:00:17.930000     1900-01-01 00:00:18.250000   
25%    1900-01-01 00:00:51.249999872     1900-01-01 00:00:51.180000   
50%       1900-01-01 00:01:42.240000  1900-01-01 00:01:42.590000128   
75%       1900-01-01 00:02:10.020000     1900-01-01 00:02:12.160000   
max       1900-01-01 00:16:21.380000     1900-01-01 00:16:32.050000   
std                              NaN                            NaN   

            Points  
count  1949.000000  
mean      3.658286  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       4.000000  
max      40.000000  
std       7.421645

In [30]:
dfResults.isna().sum()

Event                0
Category             0
Name1                0
Year1                0
Name2             1720
Year2             1720
Name3             1720
Year3             1720
Name4             1720
Year4             1720
School               0
QualifyingTime       0
Time                66
Place                0
Points               0
Division             0
dtype: int64

The qualifying and result times are now datetimes. The Time column has a few NaT entries due to DQs and DFSs, while the qualifying time does not have any since a qualifying time is required. There are also NaN's for the relay swimmer columns (name/year 2-4) for individual event entries.

In [34]:
(~dfResults.Event.str.contains("Relay")).sum()

1720

Finally, in case I want to only consider finals or prelims separately, or want to drop the non-recorded times, I will split the dataframe into a few different subsets.

In [ ]:
dfResultsFinals = dfResults[dfResults.Category.isin(
                            ['Timed Final Relay','Championship Final','Consolation Final', 'Timed Final Individual'])]
dfResultsPrelims = dfResults[dfResults.Category.isin(['Preliminaries','Swim-off'])]

In [ ]:
print(len(dfResults), len(dfResultsFinals), len(dfResultsPrelims))

In [ ]:
dfResultsDropDQ = dfResults[(dfResults.Time != 'DQ') & (dfResults.Time != 'DFS')]
dfResultsDropDQ.describe(include='all')

In [ ]:
dfResultsFinals.dtypes

In [ ]:
dfResultsFinals[~dfResultsFinals.Time.isin(['DQ','DFS'])].Place.unique()

I've now cleaned the dataframes and split the data into a few different subsets that will be useful for different analyses. I can now tackle the questions I set forth to answer.

### Analyzing Data

#### Question 1: Which team(s) did best (and how should that be defined)?

The obvious answer to this question would be the team that scored the most points in the men's and women's competitions. And if I consider those 2 parts of the same team, the team with the most combined points would be the best. These answers can essentially be read right off the results (the PDF results add diving points which I will be ignoring for this analysis of purely swimming results).

But there are a few other considerations of which teams did the best at championships. Beyond just points scored, perhaps success should be measured in terms of the number of swimmers or total number of swims at championships or in a finals heat -- the most successful teams are the teams that sent the most swimmers to National Championships. Or, perhaps the opposite should be considered -- which teams had the most points per swimmer, regardless of the total number they sent.

Alternatively, maybe success should be defined by performance at this meet relative to other meets this season. Which swimmers and teams dropped the most time at this meet?

So, I will take these questions in turn and come up with a few notions of which teams had the "best" meet.

In [36]:
print(dfResults.School.unique())
print(dfResults.School.nunique())

['NC State' 'Arizona St' 'Florida' 'California' 'Indiana' 'Tennessee'
 'Louisville' 'Virginia' 'Auburn' 'Stanford' 'Alabama' 'Texas' 'Georgia'
 'Texas A&M' 'Wisconsin' 'Ohio St' 'Notre Dame' 'Michigan' 'Minnesota'
 'Pittsburgh' 'Harvard' 'Missouri' 'Virginia Tech' 'UNC' 'Kentucky'
 'Georgia Tech' 'Yale' 'Princeton' 'Florida St' 'Purdue' 'SMU'
 'West Virginia' 'Towson' 'Brown' 'Utah' 'Brigham Young'
 'Southern California' 'LSU' 'Penn St' 'Cal Baptist' 'Arizona' 'Air Force'
 'UNLV' 'Northwestern' 'Columbia' 'TCU' 'IUPUI' 'SIUC' 'South Carolina'
 'Arkansas' 'Duke' 'UCLA' 'Miami (Ohio)' 'Penn' 'Nebraska' 'Akron'
 'Oakland' 'GWU' 'Miami (FL)' 'San Diego St' 'Hawaii' 'UNC Asheville'
 'Buffalo' 'William & Mary' 'Nevada' 'Rice' 'Denver' "Florida Int'l"
 'Cincinnati' 'Washington St.']
70


In [69]:
print(dfResults[dfResults.Division=='Women\'s DI'].School.nunique())
print(dfResults[~(dfResults.Division=='Women\'s DI')].School.nunique())
print(len(np.intersect1d(dfResults[dfResults.Division=='Women\'s DI'].School.unique(), 
                            dfResults[~(dfResults.Division=='Women\'s DI')].School.unique())))

55
49
34


There are 55 unique schools in the Women's competition and 49 for the Men's (because of the 34 schools in both there are 70 total unique schools across both meets).

In [63]:
pointsTable = dfResults[['School','Points','Division']].groupby(['School','Division']).sum().reset_index()
pointsTable.groupby('School').sum().sort_values(by='Points',ascending=False).head(10)

Points
School            
Texas        705.5
NC State     636.5
Virginia     619.5
California   616.0
Florida      527.0
Indiana      459.0
Stanford     454.5
Arizona St   449.0
Tennessee    398.5
Louisville   358.0

In [64]:
pointsTable.sort_values(by='Points', ascending=False).head(10)

School    Division  Points
95    Virginia  Women's DI   541.5
15  California    Men's DI   479.0
6   Arizona St    Men's DI   430.0
50    NC State    Men's DI   373.5
83       Texas  Women's DI   365.5
21     Florida    Men's DI   353.0
82       Texas    Men's DI   340.0
78    Stanford  Women's DI   333.0
34     Indiana    Men's DI   275.0
41  Louisville  Women's DI   266.0

In [65]:
pointsTable[pointsTable.Division=='Women\'s DI'].sort_values(by='Points', ascending=False).head(10)

School    Division  Points
95    Virginia  Women's DI   541.5
83       Texas  Women's DI   365.5
78    Stanford  Women's DI   333.0
41  Louisville  Women's DI   266.0
51    NC State  Women's DI   263.0
60     Ohio St  Women's DI   216.0
81   Tennessee  Women's DI   214.0
35     Indiana  Women's DI   184.0
22     Florida  Women's DI   174.0
16  California  Women's DI   137.0

In [66]:
pointsTable[~(pointsTable.Division=='Women\'s DI')].sort_values(by='Points', ascending=False).head(10)

School  Division  Points
15     California  Men's DI   479.0
6      Arizona St  Men's DI   430.0
50       NC State  Men's DI   373.5
21        Florida  Men's DI   353.0
82          Texas  Men's DI   340.0
34        Indiana  Men's DI   275.0
80      Tennessee  Men's DI   184.5
96  Virginia Tech  Men's DI   127.0
77       Stanford  Men's DI   121.5
9          Auburn  Men's DI   121.0

From the above, we can see the top schools by the number of points scored for a few selections. For the Women's meet the top schools were Virginia, Texas, Stanford. For the Men's the top 3 were Cal, Arizona St., and NC State. If we compare the top performing teams in the two groups, the highest scorers were the Virginia Women, Cal Men, and Arizona St. Men. Finally, when combining men's and women's scores, the top 3 schools were Texas, NC State, and Virginia.

Next, we look at the related but separate question of which schools led in terms of number of swimmers.

First, I'll only consider swimmers in individual events.

In [88]:
swimmersTable = dfResults[~dfResults.Event.str.contains("Relay")][['School','Name1','Points','Division']].groupby(
    ['School','Division']).agg({'Name1':'nunique','Points':'sum'}).reset_index()

In [89]:
#swimmersTable
swimmersTable.sort_values(by='Name1', ascending=False).head(10)

School    Division  Name1  Points
94    Virginia  Women's DI     17   341.5
50    NC State    Men's DI     17   215.5
22     Florida  Women's DI     16    78.0
21     Florida    Men's DI     16   173.0
15  California    Men's DI     16   321.0
78    Stanford  Women's DI     15   183.0
60     Ohio St  Women's DI     15    94.0
82       Texas    Men's DI     15   216.0
6   Arizona St    Men's DI     15   270.0
83       Texas  Women's DI     14   219.5

Consistency check to make sure I didn't mess anything up while grouping and summing: do these values match the points totals for teams from individual events?

In [91]:
dfResults[~dfResults.Event.str.contains("Relay")][['School','Points','Division']].groupby(
    ['School','Division']).sum().reset_index().sort_values(by='Points', ascending=False)

School    Division  Points
94     Virginia  Women's DI   341.5
15   California    Men's DI   321.0
6    Arizona St    Men's DI   270.0
83        Texas  Women's DI   219.5
82        Texas    Men's DI   216.0
..          ...         ...     ...
79          TCU    Men's DI     0.0
49     Missouri  Women's DI     0.0
33        IUPUI    Men's DI     0.0
31      Harvard    Men's DI     0.0
102        Yale    Men's DI     0.0

[103 rows x 3 columns]

Matching teams in the two previous tables, I can see that they give the same results.
Now, I'll add a points per swimmer column to the table.

In [102]:
swimmersTable['PointsPerSwimmer'] = swimmersTable.Points / swimmersTable.Name1
swimmersTable.rename(columns = {'Name1':'Swimmers'}, inplace=True)
swimmersTable.sort_values(by='PointsPerSwimmer',ascending=False).head(10)

School    Division  Swimmers  Points  PointsPerSwimmer
38         LSU    Men's DI         1    43.5         43.500000
94    Virginia  Women's DI        17   341.5         20.088235
15  California    Men's DI        16   321.0         20.062500
6   Arizona St    Men's DI        15   270.0         18.000000
56  Notre Dame    Men's DI         3    52.0         17.333333
83       Texas  Women's DI        14   219.5         15.678571
34     Indiana    Men's DI         9   139.0         15.444444
82       Texas    Men's DI        15   216.0         14.400000
39         LSU  Women's DI         4    53.0         13.250000
81   Tennessee  Women's DI        10   132.0         13.200000

When looking at the number of swimmers and points per swimmer we see lots of the same names the points table. Virginia Women did well in points, swimmmers, and points per swimmer, as did Cal and Arizona St. Men.

One standout when looking at points per swimmer is LSU Men who took 43.5 points with their only swimmer.

In [94]:
dfResults[(dfResults.School=='LSU') & (dfResults.Division=='Men\'s DI')]

Event            Category         Name1 Year1  \
188    Event 5  Men 50 Yard Freestyle  Championship Final  Brooks Curry    SR   
204    Event 5  Men 50 Yard Freestyle       Preliminaries  Brooks Curry    SR   
397  Event 10  Men 200 Yard Freestyle  Championship Final  Brooks Curry    SR   
413  Event 10  Men 200 Yard Freestyle       Preliminaries  Brooks Curry    SR   
693  Event 17  Men 100 Yard Freestyle  Championship Final  Brooks Curry    SR   
708  Event 17  Men 100 Yard Freestyle       Preliminaries  Brooks Curry    SR   

    Name2 Year2 Name3 Year3 Name4 Year4 School          QualifyingTime  \
188   NaN   NaN   NaN   NaN   NaN   NaN    LSU 1900-01-01 00:00:18.720   
204   NaN   NaN   NaN   NaN   NaN   NaN    LSU 1900-01-01 00:00:18.940   
397   NaN   NaN   NaN   NaN   NaN   NaN    LSU 1900-01-01 00:01:31.940   
413   NaN   NaN   NaN   NaN   NaN   NaN    LSU 1900-01-01 00:01:33.150   
693   NaN   NaN   NaN   NaN   NaN   NaN    LSU 1900-01-01 00:00:41.170   
708   NaN   NaN   NaN   NaN   NaN   NaN    LSU 1900-01-01 00:00:41.860   

                       Time Place  Points  Division  
188 1900-01-01 00:00:18.760     4    15.0  Men's DI  
204 1900-01-01 00:00:18.720     4     0.0  Men's DI  
397 1900-01-01 00:01:31.300     4    15.0  Men's DI  
413 1900-01-01 00:01:31.940     4     0.0  Men's DI  
693 1900-01-01 00:00:41.030     5    13.5  Men's DI  
708 1900-01-01 00:00:41.170     4     0.0  Men's DI

All of LSU Men's swims are from Brooks Curry, who totaled 43.5 points across 3 events. Other teams that picked up lots of points per swimmer but didn't have enough total swimmers to get them into the top of the team points totals include Notre Dame's Men (52 points from 3 swimmers), Indiana's Men (139 points from 9 swimmers), and LSU's Women (53 points from 4 swimmers). This also highlights Brooks's great result: compared to the LSU women, he achieved 82% of their score from 4 swimmers by himself.

This also suggests another metric: points per number of swims (rather than swimmers) or average place. This picks up teams that tended to have their swimmers score lots of points (high points finishes). I expect to find lots of the same teams, but there's also a possibility of finding something similar to the LSU/Brooks Curry finding. Perhaps there are teams that didn't necessarily get lots of swimmers into finals and score lots of points, but when they did they tended to place well.

Before that, I will take a look at teams on the other end of the previous table: those who did not score many points.

In [103]:
swimmersTable.sort_values(by=['Points','Swimmers'], ascending=[True,False])

School    Division  Swimmers  Points  PointsPerSwimmer
8         Arkansas  Women's DI         6     0.0          0.000000
74  South Carolina  Women's DI         6     0.0          0.000000
24      Florida St    Men's DI         3     0.0          0.000000
49        Missouri  Women's DI         3     0.0          0.000000
72    San Diego St  Women's DI         3     0.0          0.000000
..             ...         ...       ...     ...               ...
82           Texas    Men's DI        15   216.0         14.400000
83           Texas  Women's DI        14   219.5         15.678571
6       Arizona St    Men's DI        15   270.0         18.000000
15      California    Men's DI        16   321.0         20.062500
94        Virginia  Women's DI        17   341.5         20.088235

[103 rows x 5 columns]

The Arkansas and South Carolina Women's team managed to qualify 6 swimmers for national championships but did not score any points.

In [106]:
swimmersTable[swimmersTable.Points==0].sort_values(by='Swimmers', ascending=False).reset_index()

index               School    Division  Swimmers  Points  PointsPerSwimmer
0       8             Arkansas  Women's DI         6     0.0               0.0
1      74       South Carolina  Women's DI         6     0.0               0.0
2      49             Missouri  Women's DI         3     0.0               0.0
3      72         San Diego St  Women's DI         3     0.0               0.0
4      24           Florida St    Men's DI         3     0.0               0.0
5      57           Notre Dame  Women's DI         2     0.0               0.0
6      97       Washington St.  Women's DI         2     0.0               0.0
7      75  Southern California    Men's DI         2     0.0               0.0
8      64           Pittsburgh  Women's DI         2     0.0               0.0
9       5              Arizona  Women's DI         2     0.0               0.0
10     54         Northwestern    Men's DI         2     0.0               0.0
11    102                 Yale    Men's DI         2     0.0               0.0
12     31              Harvard    Men's DI         1     0.0               0.0
13     33                IUPUI    Men's DI         1     0.0               0.0
14     99       William & Mary  Women's DI         1     0.0               0.0
15     98        West Virginia    Men's DI         1     0.0               0.0
16     11        Brigham Young    Men's DI         1     0.0               0.0
17     91                 UNLV  Women's DI         1     0.0               0.0
18     90        UNC Asheville  Women's DI         1     0.0               0.0
19     79                  TCU    Men's DI         1     0.0               0.0
20     12                Brown    Men's DI         1     0.0               0.0
21     13              Buffalo  Women's DI         1     0.0               0.0
22     73       South Carolina    Men's DI         1     0.0               0.0
23     14          Cal Baptist    Men's DI         1     0.0               0.0
24     71                  SMU  Women's DI         1     0.0               0.0
25     70                  SMU    Men's DI         1     0.0               0.0
26     68                 Rice  Women's DI         1     0.0               0.0
27     66            Princeton  Women's DI         1     0.0               0.0
28     17           Cincinnati  Women's DI         1     0.0               0.0
29     58              Oakland  Women's DI         1     0.0               0.0
30     18             Columbia    Men's DI         1     0.0               0.0
31     19               Denver  Women's DI         1     0.0               0.0
32     53               Nevada  Women's DI         1     0.0               0.0
33     52             Nebraska  Women's DI         1     0.0               0.0
34     26                  GWU  Women's DI         1     0.0               0.0
35      4              Arizona    Men's DI         1     0.0               0.0

Overall, 36 teams ended the meet with 0 points from individual events.

In [117]:
dfTmp = swimmersTable[['School','Points']].groupby('School').sum().reset_index()
dfTmp[dfTmp.Points==0].reset_index()

index          School  Points
0       3         Arizona     0.0
1       5        Arkansas     0.0
2       7   Brigham Young     0.0
3       8           Brown     0.0
4       9         Buffalo     0.0
5      10     Cal Baptist     0.0
6      12      Cincinnati     0.0
7      13        Columbia     0.0
8      14          Denver     0.0
9      19             GWU     0.0
10     22         Harvard     0.0
11     24           IUPUI     0.0
12     35        Nebraska     0.0
13     36          Nevada     0.0
14     39         Oakland     0.0
15     46            Rice     0.0
16     48             SMU     0.0
17     49    San Diego St     0.0
18     50  South Carolina     0.0
19     53             TCU     0.0
20     60   UNC Asheville     0.0
21     61            UNLV     0.0
22     65  Washington St.     0.0
23     66   West Virginia     0.0
24     67  William & Mary     0.0
25     69            Yale     0.0

Combining Men's and Women's teams, 26 schools left the championships with no points.

Now, back to considering points per swim or average place. First I will consider total event entries per school. All entries will have a prelim entry, and those that made finals will have a repeat (except for the 1650 which has just one swim, the 'timed final'). I am only interested in counting each case as one "swimmer entry." So I want a table of total number of prelim swims per team, along with total points scored. I will also consider which schools did best at converting prelims entries into finals qualifications.

Then, I will limit the scope to only finals swim and which schools converted finals swims into the most points and the highest average finishing places.

In [293]:
#pointsTable = dfResults[~dfResults.Event.str.contains("Relay")][['School','Name1','Points','Place','Division']].groupby(
#    ['School','Division']).agg({'Name1':'sum','Points':'sum'}).reset_index()
pointsTable = dfResults[~dfResults.Event.str.contains("Relay")][
    ['School','Points','Name1','Division','Category']]
#.groupby(['School','Division']).value_counts().reset_index()
pointsTable.Category.unique()
#.agg({
#    'Name1':'nunique','Points':'sum'
#}).reset_index()

#value_counts().reset_index()

array(['Championship Final', 'Consolation Final', 'Preliminaries',
       'Timed Final Individual', 'Swim-off'], dtype=object)

"Timed Final Individual" is both a prelim and final, so I will have to count it as both. I'll eventually count every swim in this type of event as if it were in both Preliminaries and the Championship Final. This messes with the analysis a little because there will be many more than just 8 swimmers in the championship final for this event, but everyone has a chance to win the championship final points, so I think this is the best option. A swim-off is essentially a re-do of a prelim so it shouldn't really be counted here. I'll make new columns that are bools for each category that I can easily sum over.

In [294]:
pointsTable['Prelim'] = (pointsTable.Category == 'Preliminaries') | (pointsTable.Category == 'Timed Final Individual')
pointsTable['BFinal'] = pointsTable.Category == 'Consolation Final'
pointsTable['AFinal'] = pointsTable.Category == 'Championship Final'
pointsTable['TimedFinal'] = pointsTable.Category == 'Timed Final Individual'
pointsTable['BPoints'] = pointsTable[pointsTable.Category == 'Consolation Final'].Points
pointsTable['APoints'] = pointsTable[pointsTable.Category == 'Championship Final'].Points
pointsTable['TimedPoints'] = pointsTable[pointsTable.Category == 'Timed Final Individual'].Points
pointsTable

School  Points              Name1    Division            Category  \
44        Texas    20.0        Luke Hobson    Men's DI  Championship Final   
45        Texas    17.0     David Johnston    Men's DI  Championship Final   
46      Georgia    16.0       Jake Magahey    Men's DI  Championship Final   
47    Wisconsin    15.0       Jake Newmark    Men's DI  Championship Final   
48      Florida    14.0      Jake Mitchell    Men's DI  Championship Final   
...         ...     ...                ...         ...                 ...   
1918    Ohio St     0.0    Kyra Sommerstad  Women's DI       Preliminaries   
1919       UCLA     0.0   Paige MacEachern  Women's DI       Preliminaries   
1920       Duke     0.0  Catherine Purnell  Women's DI       Preliminaries   
1921       Rice     0.0      Arielle Hayon  Women's DI       Preliminaries   
1922   Kentucky     0.0       Lauren Poole  Women's DI       Preliminaries   

      Prelim  BFinal  AFinal  TimedFinal  BPoints  APoints  TimedPoints  
44     False   False    True       False      NaN     20.0          NaN  
45     False   False    True       False      NaN     17.0          NaN  
46     False   False    True       False      NaN     16.0          NaN  
47     False   False    True       False      NaN     15.0          NaN  
48     False   False    True       False      NaN     14.0          NaN  
...      ...     ...     ...         ...      ...      ...          ...  
1918    True   False   False       False      NaN      NaN          NaN  
1919    True   False   False       False      NaN      NaN          NaN  
1920    True   False   False       False      NaN      NaN          NaN  
1921    True   False   False       False      NaN      NaN          NaN  
1922    True   False   False       False      NaN      NaN          NaN  

[1720 rows x 12 columns]

In [295]:
pointsTable = pointsTable.groupby(['School','Division']).sum().reset_index()
pointsTable

School    Division  Points  Prelim  BFinal  AFinal  TimedFinal  \
0         Air Force    Men's DI     2.0       3       1       0           0   
1             Akron  Women's DI     5.0       9       2       0           0   
2           Alabama    Men's DI    21.0      13       2       1           1   
3           Alabama  Women's DI    73.0      20       4       2           1   
4           Arizona    Men's DI     0.0       3       0       0           0   
..              ...         ...     ...     ...     ...     ...         ...   
98    West Virginia    Men's DI     0.0       3       0       0           0   
99   William & Mary  Women's DI     0.0       3       0       0           0   
100       Wisconsin    Men's DI    19.0      11       1       1           0   
101       Wisconsin  Women's DI    82.0      21       1       4           2   
102            Yale    Men's DI     0.0       5       0       0           1   

     BPoints  APoints  TimedPoints  
0        2.0      0.0          0.0  
1        5.0      0.0          0.0  
2        8.0     11.0          2.0  
3       22.0     31.0         20.0  
4        0.0      0.0          0.0  
..       ...      ...          ...  
98       0.0      0.0          0.0  
99       0.0      0.0          0.0  
100      4.0     15.0          0.0  
101      3.0     63.0         16.0  
102      0.0      0.0          0.0  

[103 rows x 10 columns]

Now I have a table that has a row for each team with the total points scored and number of swims in each category of prelim/final as well as how many points each team got from each type of final.

In [296]:
pointsTable['pointsPerPrelim'] = pointsTable.Points / (pointsTable.Prelim + pointsTable.TimedFinal)
pointsTable['pointsPerFinal'] = pointsTable.Points / (pointsTable.BFinal + pointsTable.AFinal + pointsTable.TimedFinal)
pointsTable['BPointsPercentage'] = pointsTable.BPoints / pointsTable.Points
pointsTable['APointsPercentage'] = pointsTable.APoints / pointsTable.Points
pointsTable['TimedPointsPercentage'] = pointsTable.TimedPoints / pointsTable.Points
pointsTable

School    Division  Points  Prelim  BFinal  AFinal  TimedFinal  \
0         Air Force    Men's DI     2.0       3       1       0           0   
1             Akron  Women's DI     5.0       9       2       0           0   
2           Alabama    Men's DI    21.0      13       2       1           1   
3           Alabama  Women's DI    73.0      20       4       2           1   
4           Arizona    Men's DI     0.0       3       0       0           0   
..              ...         ...     ...     ...     ...     ...         ...   
98    West Virginia    Men's DI     0.0       3       0       0           0   
99   William & Mary  Women's DI     0.0       3       0       0           0   
100       Wisconsin    Men's DI    19.0      11       1       1           0   
101       Wisconsin  Women's DI    82.0      21       1       4           2   
102            Yale    Men's DI     0.0       5       0       0           1   

     BPoints  APoints  TimedPoints  pointsPerPrelim  pointsPerFinal  \
0        2.0      0.0          0.0         0.666667        2.000000   
1        5.0      0.0          0.0         0.555556        2.500000   
2        8.0     11.0          2.0         1.500000        5.250000   
3       22.0     31.0         20.0         3.476190       10.428571   
4        0.0      0.0          0.0         0.000000             NaN   
..       ...      ...          ...              ...             ...   
98       0.0      0.0          0.0         0.000000             NaN   
99       0.0      0.0          0.0         0.000000             NaN   
100      4.0     15.0          0.0         1.727273        9.500000   
101      3.0     63.0         16.0         3.565217       11.714286   
102      0.0      0.0          0.0         0.000000        0.000000   

     BPointsPercentage  APointsPercentage  TimedPointsPercentage  
0             1.000000           0.000000               0.000000  
1             1.000000           0.000000               0.000000  
2             0.380952           0.523810               0.095238  
3             0.301370           0.424658               0.273973  
4                  NaN                NaN                    NaN  
..                 ...                ...                    ...  
98                 NaN                NaN                    NaN  
99                 NaN                NaN                    NaN  
100           0.210526           0.789474               0.000000  
101           0.036585           0.768293               0.195122  
102                NaN                NaN                    NaN  

[103 rows x 15 columns]

In [297]:
pointsTable[['School','Division','Points','Prelim','pointsPerPrelim']].sort_values(by='pointsPerPrelim', ascending=False).head(10)

School    Division  Points  Prelim  pointsPerPrelim
38         LSU    Men's DI    43.5       3        14.500000
15  California    Men's DI   321.0      43         7.295455
94    Virginia  Women's DI   341.5      48         6.830000
69        SIUC    Men's DI    13.0       2         6.500000
83       Texas  Women's DI   219.5      32         6.271429
34     Indiana    Men's DI   139.0      23         6.043478
56  Notre Dame    Men's DI    52.0       8         5.777778
6   Arizona St    Men's DI   270.0      45         5.625000
82       Texas    Men's DI   216.0      39         5.142857
35     Indiana  Women's DI   102.0      19         4.857143

At the top of the points per prelim metric we see the LSU Men's team again, who scored 43.5 points off of 3 Brooks Curry entries for 14.5 points per prelim. This is nearly twice as much as the next nearest team. Cal's Men's team managed to score an impressive 7.3 points per prelim with 43 entries. The Virginia Women round out the top 3 with 6.8 points per prelim and 48 total prelims. Of note in this top 10 are 2 smaller teams who managed to perform well: SIUC's Men's Team (2 entries, 6.5 points per prelim) and Notre Dame's Men's Team (8 entries, 5.8 points per prelim).

In [229]:
dfResults[(dfResults.School=='SIUC') & (dfResults.Division=='Men\'s DI')]

Event           Category            Name1  \
526  Event 12  Men 100 Yard Backstroke  Consolation Final  Ruard Van Renen   
543  Event 12  Men 100 Yard Backstroke      Preliminaries  Ruard Van Renen   
645  Event 16  Men 200 Yard Backstroke  Consolation Final  Ruard Van Renen   
658  Event 16  Men 200 Yard Backstroke      Preliminaries  Ruard Van Renen   

    Year1 Name2 Year2 Name3 Year3 Name4 Year4 School          QualifyingTime  \
526    FR   NaN   NaN   NaN   NaN   NaN   NaN   SIUC 1900-01-01 00:00:45.170   
543    FR   NaN   NaN   NaN   NaN   NaN   NaN   SIUC 1900-01-01 00:00:44.890   
645    FR   NaN   NaN   NaN   NaN   NaN   NaN   SIUC 1900-01-01 00:01:39.730   
658    FR   NaN   NaN   NaN   NaN   NaN   NaN   SIUC 1900-01-01 00:01:40.640   

                       Time Place  Points  Division  
526 1900-01-01 00:00:44.670     9     9.0  Men's DI  
543 1900-01-01 00:00:45.170    10     0.0  Men's DI  
645 1900-01-01 00:01:40.220    13     4.0  Men's DI  
658 1900-01-01 00:01:39.730    10     0.0  Men's DI

Ruard Van Renen from SIUC was the only SIUC swimmer at the meet, making 2 Consolation Finals, scoring a total of 13 points by himself. (This strong peformance also helps highlight once again how impressive Brooks Curry's 14.5 points per event as the only LSU Men's entry is.)

Now I'll consider teams that were able to turn prelim swims into finals entries.

In [298]:
pointsTable.assign(FinalPercentage = 
                   (pointsTable.AFinal + pointsTable.BFinal + pointsTable.TimedFinal)/pointsTable.Prelim,
                  Final = pointsTable.AFinal + pointsTable.BFinal + pointsTable.TimedFinal)[
    ['School','Division','Points','Prelim','Final','FinalPercentage']].sort_values(
    by='FinalPercentage',ascending=False).drop('FinalPercentage',axis=1).reset_index(drop=True).head(10)

School    Division  Points  Prelim  Final
0        SIUC    Men's DI    13.0       2      2
1         LSU    Men's DI    43.5       3      3
2  Notre Dame    Men's DI    52.0       8      7
3     Indiana  Women's DI   102.0      19     12
4  California    Men's DI   321.0      43     26
5  Arizona St    Men's DI   270.0      45     27
6    Virginia  Women's DI   341.5      48     28
7       Texas  Women's DI   219.5      32     18
8  Louisville  Women's DI   140.0      28     15
9    NC State  Women's DI   155.0      34     18

We see lots of repeat top performing teams here, many teams in the top 10 have been discussed as they have been in other top 10s. The top 10 teams below Indiana's Women at #4 (12 of 17 entries in finals) are all large teams that had many swimmer in prelims and finals and took home large points hauls. As discussed, SIUC and LSU's swimmers made finals in all their events. Notre Dame's Men's Team converted 7 of 8 swims into finals (6 consolation finals + 1 timed 1650 final). The lone event they missed the finals of was the 50 yard freestyle, where Chris Guiliano finished 23rd in the prelims. In the 100 Backstroke prelims, Tommy Janton finished tied for 16th so had to go to a swim-off, which he won to earn the last spot in the consolation final.

In [237]:
dfResults[(dfResults.School=='Notre Dame') & (dfResults.Division=='Men\'s DI') & 
          (dfResults.Category!='Timed Final Relay')]

Event                Category  \
53               Event 3  Men 500 Yard Freestyle       Consolation Final   
72               Event 3  Men 500 Yard Freestyle           Preliminaries   
223               Event 5  Men 50 Yard Freestyle           Preliminaries   
290                     Event 8  Men 400 Yard IM       Consolation Final   
310                     Event 8  Men 400 Yard IM           Preliminaries   
402             Event 10  Men 200 Yard Freestyle       Consolation Final   
419             Event 10  Men 200 Yard Freestyle           Preliminaries   
531            Event 12  Men 100 Yard Backstroke       Consolation Final   
549            Event 12  Men 100 Yard Backstroke           Preliminaries   
603            Event 15  Men 1650 Yard Freestyle  Timed Final Individual   
642            Event 16  Men 200 Yard Backstroke       Consolation Final   
660            Event 16  Men 200 Yard Backstroke           Preliminaries   
697             Event 17  Men 100 Yard Freestyle       Consolation Final   
712             Event 17  Men 100 Yard Freestyle           Preliminaries   
910  Event 112  Men 100 Yard Backstroke Swim-off                Swim-off   

              Name1 Year1 Name2 Year2 Name3 Year3 Name4 Year4      School  \
53    Jack Hoagland    SR   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
72    Jack Hoagland    SR   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
223  Chris Guiliano    SO   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
290   Jack Hoagland    SR   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
310   Jack Hoagland    SR   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
402  Chris Guiliano    SO   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
419  Chris Guiliano    SO   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
531    Tommy Janton    FR   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
549    Tommy Janton    FR   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
603   Jack Hoagland    SR   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
642    Tommy Janton    FR   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
660    Tommy Janton    FR   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
697  Chris Guiliano    SO   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
712  Chris Guiliano    SO   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   
910    Tommy Janton    FR   NaN   NaN   NaN   NaN   NaN   NaN  Notre Dame   

             QualifyingTime                    Time Place  Points  Division  
53  1900-01-01 00:04:12.770 1900-01-01 00:04:12.490    10     7.0  Men's DI  
72  1900-01-01 00:04:14.240 1900-01-01 00:04:12.770    13     0.0  Men's DI  
223 1900-01-01 00:00:18.880 1900-01-01 00:00:19.170    23     0.0  Men's DI  
290 1900-01-01 00:03:41.670 1900-01-01 00:03:40.820    12     5.0  Men's DI  
310 1900-01-01 00:03:41.150 1900-01-01 00:03:41.670    16     0.0  Men's DI  
402 1900-01-01 00:01:32.360 1900-01-01 00:01:32.310     9     9.0  Men's DI  
419 1900-01-01 00:01:32.430 1900-01-01 00:01:32.360    10     0.0  Men's DI  
531 1900-01-01 00:00:45.540 1900-01-01 00:00:45.430    14     3.0  Men's DI  
549 1900-01-01 00:00:45.610 1900-01-01 00:00:45.540    16     0.0  Men's DI  
603 1900-01-01 00:14:48.820 1900-01-01 00:14:38.640     5    14.0  Men's DI  
642 1900-01-01 00:01:39.830 1900-01-01 00:01:39.450    10     7.0  Men's DI  
660 1900-01-01 00:01:40.300 1900-01-01 00:01:39.830    12     0.0  Men's DI  
697 1900-01-01 00:00:41.480 1900-01-01 00:00:41.640    10     7.0  Men's DI  
712 1900-01-01 00:00:41.850 1900-01-01 00:00:41.480     9     0.0  Men's DI  
910 1900-01-01 00:00:45.540 1900-01-01 00:00:45.120     1     0.0  Men's DI

We can also consider which teams failed to make finals, despite many prelims entries.

In [300]:
dfTmp = pointsTable.assign(FinalPercentage = 
                   (pointsTable.AFinal + pointsTable.BFinal + pointsTable.TimedFinal)/pointsTable.Prelim,
                  Final = pointsTable.AFinal + pointsTable.BFinal + pointsTable.TimedFinal)[
    ['School','Division','Points','Prelim','Final','FinalPercentage']].sort_values(
    by=['FinalPercentage','Prelim'],ascending=[True,False]).drop('FinalPercentage',axis=1).reset_index(drop=True)
dfTmp[dfTmp.Final==0]

School    Division  Points  Prelim  Final
0        South Carolina  Women's DI     0.0      15      0
1              Arkansas  Women's DI     0.0      14      0
2              Missouri  Women's DI     0.0       7      0
3          San Diego St  Women's DI     0.0       7      0
4               Arizona  Women's DI     0.0       6      0
5            Notre Dame  Women's DI     0.0       5      0
6            Pittsburgh  Women's DI     0.0       5      0
7   Southern California    Men's DI     0.0       4      0
8               Arizona    Men's DI     0.0       3      0
9         Brigham Young    Men's DI     0.0       3      0
10                Brown    Men's DI     0.0       3      0
11          Cal Baptist    Men's DI     0.0       3      0
12               Denver  Women's DI     0.0       3      0
13                  GWU  Women's DI     0.0       3      0
14         Northwestern    Men's DI     0.0       3      0
15              Oakland  Women's DI     0.0       3      0
16                  SMU    Men's DI     0.0       3      0
17       Washington St.  Women's DI     0.0       3      0
18        West Virginia    Men's DI     0.0       3      0
19       William & Mary  Women's DI     0.0       3      0
20              Buffalo  Women's DI     0.0       2      0
21             Columbia    Men's DI     0.0       2      0
22              Harvard    Men's DI     0.0       2      0
23                IUPUI    Men's DI     0.0       2      0
24            Princeton  Women's DI     0.0       2      0
25                 Rice  Women's DI     0.0       2      0
26                  SMU  Women's DI     0.0       2      0
27        UNC Asheville  Women's DI     0.0       2      0
28                 UNLV  Women's DI     0.0       2      0
29           Cincinnati  Women's DI     0.0       1      0
30               Nevada  Women's DI     0.0       1      0
31       South Carolina    Men's DI     0.0       1      0
32                  TCU    Men's DI     0.0       1      0

Despite qualifying a lot of swimmers for national championships, South Carolina Women and Arkansas Women were unable to convert any of their 15 and 14 prelim swims, respectively, into finals appearances. These 2 programs had twice as many as any other teams that secured 0 inividual finals.

In [301]:
dfResults[(dfResults.School.isin(['South Carolina','Arkansas'])) & (dfResults.Division=='Women\'s DI') & 
          (dfResults.Category!='Timed Final Relay')]

Event       Category  \
1011      Event 3  Women 500 Yard Freestyle  Preliminaries   
1078             Event 4  Women 200 Yard IM  Preliminaries   
1108             Event 4  Women 200 Yard IM  Preliminaries   
1166       Event 5  Women 50 Yard Freestyle  Preliminaries   
1167       Event 5  Women 50 Yard Freestyle  Preliminaries   
1193       Event 5  Women 50 Yard Freestyle  Preliminaries   
1194       Event 5  Women 50 Yard Freestyle  Preliminaries   
1255             Event 8  Women 400 Yard IM  Preliminaries   
1265             Event 8  Women 400 Yard IM  Preliminaries   
1322      Event 9  Women 100 Yard Butterfly  Preliminaries   
1323      Event 9  Women 100 Yard Butterfly  Preliminaries   
1343      Event 9  Women 100 Yard Butterfly  Preliminaries   
1416     Event 10  Women 200 Yard Freestyle  Preliminaries   
1424     Event 10  Women 200 Yard Freestyle  Preliminaries   
1472  Event 11  Women 100 Yard Breaststroke  Preliminaries   
1473  Event 11  Women 100 Yard Breaststroke  Preliminaries   
1537    Event 12  Women 100 Yard Backstroke  Preliminaries   
1564    Event 12  Women 100 Yard Backstroke  Preliminaries   
1693    Event 16  Women 200 Yard Backstroke  Preliminaries   
1748     Event 17  Women 100 Yard Freestyle  Preliminaries   
1750     Event 17  Women 100 Yard Freestyle  Preliminaries   
1819  Event 18  Women 200 Yard Breaststroke  Preliminaries   
1831  Event 18  Women 200 Yard Breaststroke  Preliminaries   
1837  Event 18  Women 200 Yard Breaststroke  Preliminaries   
1903     Event 19  Women 200 Yard Butterfly  Preliminaries   
1910     Event 19  Women 200 Yard Butterfly  Preliminaries   
1911     Event 19  Women 200 Yard Butterfly  Preliminaries   
1912     Event 19  Women 200 Yard Butterfly  Preliminaries   
1913     Event 19  Women 200 Yard Butterfly  Preliminaries   

                            Name1 Year1 Name2 Year2 Name3 Year3 Name4 Year4  \
1011               Luciana Thomas    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1078                Victoria Kwan    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1108               Laura Goettler    JR   NaN   NaN   NaN   NaN   NaN   NaN   
1166  Andrea Sansores De La Fuent    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1167                  Janie Smith    SR   NaN   NaN   NaN   NaN   NaN   NaN   
1193                 Kobie Melton    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1194            Alessia Ferraguti    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1255                Victoria Kwan    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1265               Laura Goettler    JR   NaN   NaN   NaN   NaN   NaN   NaN   
1322                 Nicholle Toh    JR   NaN   NaN   NaN   NaN   NaN   NaN   
1323  Andrea Sansores De La Fuent    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1343                 Greta Pelzek    SO   NaN   NaN   NaN   NaN   NaN   NaN   
1416                  Janie Smith    SR   NaN   NaN   NaN   NaN   NaN   NaN   
1424                 Betsy Wizard    FR   NaN   NaN   NaN   NaN   NaN   NaN   
1472            Alessia Ferraguti    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1473             Vanessa Herrmann    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1537                 Kobie Melton    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1564             Tatiana Salcutan    JR   NaN   NaN   NaN   NaN   NaN   NaN   
1693             Tatiana Salcutan    JR   NaN   NaN   NaN   NaN   NaN   NaN   
1748                 Kobie Melton    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1750                  Janie Smith    SR   NaN   NaN   NaN   NaN   NaN   NaN   
1819            Alessia Ferraguti    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1831               Laura Goettler    JR   NaN   NaN   NaN   NaN   NaN   NaN   
1837             Vanessa Herrmann    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1903                 Betsy Wizard    FR   NaN   NaN   NaN   NaN   NaN   NaN   
1910                Victoria Kwan    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1911                 Nicholle Toh    JR   Na

We can repeat, only considering the Championship Final.

In [238]:
pointsTable.assign(AFinalPercentage = pointsTable.AFinal / pointsTable.Prelim)[
    ['School','Division','Points','Prelim','AFinal','AFinalPercentage']].sort_values(
    by='AFinalPercentage',ascending=False).drop('AFinalPercentage',axis=1).head(10)

School    Division  Points  Prelim  AFinal
38         LSU    Men's DI    43.5       3       3
15  California    Men's DI   321.0      43      18
94    Virginia  Women's DI   341.5      48      19
83       Texas  Women's DI   219.5      32      12
32      Hawaii  Women's DI    11.5       3       1
47   Minnesota  Women's DI    13.0       3       1
82       Texas    Men's DI   216.0      39      12
34     Indiana    Men's DI   139.0      23       7
6   Arizona St    Men's DI   270.0      45      13
41  Louisville  Women's DI   140.0      28       8

Among familiar names and impressive results from larger teams, there are 2 teams that got 1 Championship Final entry from just 3 prelim entries: Hawaii Women (Laticia-Leigh Tansom, 7th in 100 free) and Minnesota Women (Megan Van Berkom, 6th in 400 IM).

In [243]:
dfResults[(dfResults.School.isin(['Minnesota','Hawaii'])) & (dfResults.Division=='Women\'s DI')
         & ~(dfResults.Event.str.contains('Relay'))]

Event            Category  \
1091          Event 4  Women 200 Yard IM       Preliminaries   
1161    Event 5  Women 50 Yard Freestyle       Preliminaries   
1228          Event 8  Women 400 Yard IM  Championship Final   
1246          Event 8  Women 400 Yard IM       Preliminaries   
1388  Event 10  Women 200 Yard Freestyle       Preliminaries   
1708  Event 17  Women 100 Yard Freestyle  Championship Final   
1724  Event 17  Women 100 Yard Freestyle       Preliminaries   
1904  Event 19  Women 200 Yard Butterfly       Preliminaries   

                      Name1 Year1 Name2 Year2 Name3 Year3 Name4 Year4  \
1091       Megan Van Berkom    JR   NaN   NaN   NaN   NaN   NaN   NaN   
1161  Laticia-Leigh Transom    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1228       Megan Van Berkom    JR   NaN   NaN   NaN   NaN   NaN   NaN   
1246       Megan Van Berkom    JR   NaN   NaN   NaN   NaN   NaN   NaN   
1388  Laticia-Leigh Transom    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1708  Laticia-Leigh Transom    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1724  Laticia-Leigh Transom    5Y   NaN   NaN   NaN   NaN   NaN   NaN   
1904       Megan Van Berkom    JR   NaN   NaN   NaN   NaN   NaN   NaN   

         School          QualifyingTime                    Time Place  Points  \
1091  Minnesota 1900-01-01 00:01:55.670 1900-01-01 00:01:57.470    35     0.0   
1161     Hawaii 1900-01-01 00:00:22.260 1900-01-01 00:00:22.180    24     0.0   
1228  Minnesota 1900-01-01 00:04:06.700 1900-01-01 00:04:05.370     6    13.0   
1246  Minnesota 1900-01-01 00:04:04.860 1900-01-01 00:04:06.700     8     0.0   
1388     Hawaii 1900-01-01 00:01:44.890 1900-01-01 00:01:44.990    22     0.0   
1708     Hawaii 1900-01-01 00:00:47.390 1900-01-01 00:00:47.500     7    11.5   
1724     Hawaii 1900-01-01 00:00:47.860 1900-01-01 00:00:47.390     8     0.0   
1904  Minnesota 1900-01-01 00:01:55.140 1900-01-01 00:01:56.370    33     0.0   

        Division  
1091  Women's DI  
1161  Women's DI  
1228  Women's DI  
1246  Women's DI  
1388  Women's DI  
1708  Women's DI  
1724  Women's DI  
1904  Women's DI

In [244]:
pointsTable

School    Division  Points  Prelim  BFinal  AFinal  TimedFinal  \
0         Air Force    Men's DI     2.0       3       1       0           0   
1             Akron  Women's DI     5.0       9       2       0           0   
2           Alabama    Men's DI    21.0      13       2       1           1   
3           Alabama  Women's DI    73.0      20       4       2           1   
4           Arizona    Men's DI     0.0       3       0       0           0   
..              ...         ...     ...     ...     ...     ...         ...   
98    West Virginia    Men's DI     0.0       3       0       0           0   
99   William & Mary  Women's DI     0.0       3       0       0           0   
100       Wisconsin    Men's DI    19.0      11       1       1           0   
101       Wisconsin  Women's DI    82.0      21       1       4           2   
102            Yale    Men's DI     0.0       5       0       0           1   

     BPoints  APoints  TimedPoints  pointsPerPrelim  pointsPerFinal  \
0        2.0      0.0          0.0         0.666667        2.000000   
1        5.0      0.0          0.0         0.555556        2.500000   
2        8.0     11.0          2.0         1.500000        5.250000   
3       22.0     31.0         20.0         3.476190       10.428571   
4        0.0      0.0          0.0         0.000000             NaN   
..       ...      ...          ...              ...             ...   
98       0.0      0.0          0.0         0.000000             NaN   
99       0.0      0.0          0.0         0.000000             NaN   
100      4.0     15.0          0.0         1.727273        9.500000   
101      3.0     63.0         16.0         3.565217       11.714286   
102      0.0      0.0          0.0         0.000000        0.000000   

     BPointsPercentage  APointsPercentage  TimedPointsPercentage  
0             1.000000           0.000000               0.000000  
1             1.000000           0.000000               0.000000  
2             0.380952           0.523810               0.095238  
3             0.301370           0.424658               0.273973  
4                  NaN                NaN                    NaN  
..                 ...                ...                    ...  
98                 NaN                NaN                    NaN  
99                 NaN                NaN                    NaN  
100           0.210526           0.789474               0.000000  
101           0.036585           0.768293               0.195122  
102                NaN                NaN                    NaN  

[103 rows x 15 columns]

In [285]:
ATable = dfResults[~(dfResults.Event.str.contains("Relay")) & 
                   (dfResults.Category.isin(['Championship Final','Consolation Final','Timed Final Individual']))][
    ['School','Name1','Points','Division','Category','Place']]
ATable

School             Name1  Points    Division            Category  \
44           Texas       Luke Hobson    20.0    Men's DI  Championship Final   
45           Texas    David Johnston    17.0    Men's DI  Championship Final   
46         Georgia      Jake Magahey    16.0    Men's DI  Championship Final   
47       Wisconsin      Jake Newmark    15.0    Men's DI  Championship Final   
48         Florida     Jake Mitchell    14.0    Men's DI  Championship Final   
...            ...               ...     ...         ...                 ...   
1867           UNC     Ellie VanNote     5.0  Women's DI   Consolation Final   
1868    Florida St   Edith Jernstedt     4.0  Women's DI   Consolation Final   
1869      NC State        Abby Arens     3.0  Women's DI   Consolation Final   
1870  Northwestern    Miriam Guevara     2.0  Women's DI   Consolation Final   
1871       Georgia  Callie Dickinson     1.0  Women's DI   Consolation Final   

     Place  
44       1  
45       2  
46       3  
47       4  
48       5  
...    ...  
1867    12  
1868    13  
1869    14  
1870    15  
1871    16  

[459 rows x 6 columns]

In [289]:
ATable.Place = ATable.Place.astype('int')
ATable.groupby(['School','Division']).agg({
    'Name1': 'count', 'Points':'sum', 'Place':'mean'
}).sort_values(by=['Place','Name1'], ascending=[True,False]).reset_index().head(10)

School    Division  Name1  Points     Place
0            LSU  Women's DI      3    53.0  2.000000
1            LSU    Men's DI      3    43.5  4.333333
2           Utah    Men's DI      1    14.0  5.000000
3        Indiana    Men's DI     11   139.0  6.000000
4      Minnesota  Women's DI      1    13.0  6.000000
5     California    Men's DI     26   321.0  6.192308
6        Georgia    Men's DI      6    70.0  6.666667
7         Hawaii  Women's DI      1    11.5  7.000000
8  Virginia Tech    Men's DI      5    59.0  7.200000
9       Virginia  Women's DI     28   341.5  7.285714

The above table gives teams with the lowest average place, among entries in Final heats. The LSU Women and Men both had 3 finals swims, with an average finsh of 2nd place for the Women and 4.33 for the Men. Utah Men had 1 finals swim and placed 5th. We can see that in this ranking, which rewards the highest averages, the top 10 is mostly filled with teams that only had a few finals swims. Among the top 10, the teams that had both a large number of finals swims and a high average finish were Indiana Men (11 swims, 6.0 average), Cal Men (26 swims, 6.2 average), and Virginia Women (28 swims, 7.3 average). The teams that scored the most swimming points (Cal Men and Virginia Women) got lots of swimmers into finals and they finished highly on average.

In [290]:
ATable.groupby(['School','Division']).agg({
    'Name1': 'count', 'Points':'mean', 'Place':'mean'
}).sort_values(by=['Points','Name1'], ascending=[False,False]).reset_index().head(10)

School    Division  Name1     Points     Place
0            LSU  Women's DI      3  17.666667  2.000000
1            LSU    Men's DI      3  14.500000  4.333333
2           Utah    Men's DI      1  14.000000  5.000000
3      Minnesota  Women's DI      1  13.000000  6.000000
4        Indiana    Men's DI     11  12.636364  6.000000
5     California    Men's DI     26  12.346154  6.192308
6       Virginia  Women's DI     28  12.196429  7.285714
7          Texas  Women's DI     18  12.194444  7.666667
8  Virginia Tech    Men's DI      5  11.800000  7.200000
9      Wisconsin  Women's DI      7  11.714286  8.571429

This table shows how the average finish converts into points. The top 10 is mostly the same when looking at average points or average finishing place; however, because points are not perfectly linear with place (the gap in points awarded to finishing places is different for different places from 1-16) they are not exactly the same. It matters exactly how you end up with a certain average place. For example, Minnesota's Women and Indiana's Men both had an average finish of 6th. However, Minnesota had more points on average per swim (just 1 swim for Minnesota compared to 11 for Indiana). The table below shows all 12 of these results. Across the 11 swims, Indiana on average placed 6th but because of the points allocations, the smaller points awarded for finishes below 6th couldn't balance out the larger rewards for top 6 finishes, so the average points per swim was smaller than the points awarded for a single 6th place.

In [292]:
dfResults[~(dfResults.Event.str.contains("Relay")) & 
                   (dfResults.Category.isin(['Championship Final','Consolation Final','Timed Final Individual']))
         & (((dfResults.School=='Minnesota') & (dfResults.Division=='Women\'s DI')) |
            ((dfResults.School=='Indiana') & (dfResults.Division=='Men\'s DI')))]

Event            Category  \
195        Event 5  Men 50 Yard Freestyle   Consolation Final   
334       Event 9  Men 100 Yard Butterfly  Championship Final   
340       Event 9  Men 100 Yard Butterfly   Consolation Final   
406      Event 10  Men 200 Yard Freestyle   Consolation Final   
457   Event 11  Men 100 Yard Breaststroke  Championship Final   
464   Event 11  Men 100 Yard Breaststroke   Consolation Final   
518     Event 12  Men 100 Yard Backstroke  Championship Final   
694      Event 17  Men 100 Yard Freestyle  Championship Final   
769   Event 18  Men 200 Yard Breaststroke  Championship Final   
829      Event 19  Men 200 Yard Butterfly  Championship Final   
832      Event 19  Men 200 Yard Butterfly  Championship Final   
1228           Event 8  Women 400 Yard IM  Championship Final   

                 Name1 Year1 Name2 Year2 Name3 Year3 Name4 Year4     School  \
195        Van Mathias    5Y   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
334      Tomer Frankel    JR   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
340      Brendan Burns    SR   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
406    Rafael Miroslaw    SO   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
457        Van Mathias    5Y   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
464       Josh Matheny    SO   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
518      Brendan Burns    SR   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
694        Van Mathias    5Y   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
769       Josh Matheny    SO   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
829      Brendan Burns    SR   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
832      Tomer Frankel    JR   NaN   NaN   NaN   NaN   NaN   NaN    Indiana   
1228  Megan Van Berkom    JR   NaN   NaN   NaN   NaN   NaN   NaN  Minnesota   

              QualifyingTime                    Time Place  Points    Division  
195  1900-01-01 00:00:18.890 1900-01-01 00:00:18.910    11     6.0    Men's DI  
334  1900-01-01 00:00:44.260 1900-01-01 00:00:44.040     3    16.0    Men's DI  
340  1900-01-01 00:00:44.790 1900-01-01 00:00:44.600     9     9.0    Men's DI  
406  1900-01-01 00:01:32.280 1900-01-01 00:01:32.650    13     4.0    Men's DI  
457  1900-01-01 00:00:50.570 1900-01-01 00:00:50.600     2    17.0    Men's DI  
464  1900-01-01 00:00:51.170 1900-01-01 00:00:50.990     9     9.0    Men's DI  
518  1900-01-01 00:00:44.280 1900-01-01 00:00:43.610     1    20.0    Men's DI  
694  1900-01-01 00:00:41.330 1900-01-01 00:00:41.390     7    12.0    Men's DI  
769  1900-01-01 00:01:51.240 1900-01-01 00:01:50.120     4    15.0    Men's DI  
829  1900-01-01 00:01:40.510 1900-01-01 00:01:38.970     2    17.0    Men's DI  
832  1900-01-01 00:01:40.940 1900-01-01 00:01:40.340     5    14.0    Men's DI  
1228 1900-01-01 00:04:06.700 1900-01-01 00:04:05.370     6    13.0  Women's DI

Alternatively, maybe success should be defined by performance at this meet relative to other meets this season. Which swimmers and teams dropped the most time at this meet?